python/lsst/analysis/ap/nb_utils.py

```diff
@@ -70,3 +80,203 @@ def make_simbad_link(ra, dec, radius_arcsec=3.0):
         print(f"No matched sources within {radius_arcsec} arcseconds.")
 
         return None
+
+
+def compareDB(repo1, repo2, coll1, coll2, query1, query2,
+              diffimg_type1='goodSeeingDiff', diffimg_type2='goodSeeingDiff',
+              badFlagList=None, matchRadius=0.1, displayCutouts=False,
+              cutoutPath=None, chunk_size=10000,
+              add_cutout_metadata=True, njobs=1):
+    """ compare two APDB datasets by extracting unassociated sources,
+    spatially crossmatching, and plotting cutouts of the differences.
+
+    Parameters
+    ----------
+
+    repo1: `str`
+        Path to `lsst.daf.butler` repo containing the first dataset.
+        Can be the same as repo2.
+    repo2: `lsst.daf.butler`
+        Path to `lsst.daf.butler` repo containing the second dataset.
+        Can be the same as repo1.
+    coll1: `list` of `str`
+        Collection name(s) for the first dataset
+    coll2: `list` of `str`
+        Collection name(s) for the second dataset
```

Comment:
I'll say here that I think we should stop using "repo/collection" like this, and always just pass in the configured butlers themselves. There may be other necessary configurations (e.g. instrument, skymap), and it simplifies the interface.